src/Config/ConfigManager.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config;
+
+use Glpi\Application\View\TemplateRenderer;
+
+/**
+ * Centrally manages the configuration options across GLPI and plugins.
+ * This includes global options, entity options, preferences, etc.
+ */
+final class ConfigManager
+{
+    private static self $instance;
+    /** @var ConfigSection[] */
+    private static array $sections = [];
+    /** @var ConfigOption[] */
+    private static array $options = [];
+
+    /**
+     * Static constructor to initialize the sections and options arrays with the default values.
+     * @return void
+     */
+    private static function init(): void
+    {
+        if (!empty(self::$options)) {
+            return;
+        }
+        self::$sections = CoreConfigProvider::getInstance()->getConfigSections();
+        $default_options = CoreConfigProvider::getInstance()->getConfigOptions();
+        foreach ($default_options as $opt) {
+            self::getInstance()->registerOption($opt);
+        }
+    }
```

Comment:
I don't think that is possible without more changes.
Changes cause a segfault on my end. Most likely because creating the ConfigManager instance now results in a call:
`ConfigManage::__construct` -> `CoreConfigProvider::getConfigOptions` -> `ConfigManager::getInstance` ->  `ConfigManager::__construct`

Issue could be avoided by always using the section name when creating the options instead of passing a reference to the section object (something to work on tomorrow).